### PR TITLE
Add migration for existing scopes in tenant-conf.json

### PR DESCRIPTION
**Pull request by msm1992**
## Purpose
In case a scope's role mappings had been updated over the versions, we need to add the correct scope mapping in the tenant-conf.json.

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: migration-client/wso2-api-migration-client/src/main/java/org/wso2/carbon/apimgt/migration/client/ScopeRoleMappingPopulationClient.java [hunk 1]
```diff
@@ -363,4 +383,40 @@ private String makePermissionsStringByEscapingSlash(String startPermission, Stri
...
+     */
+    private static JSONObject getTenantConfFromFile() throws IOException, APIMigrationException {
+        JSONObject tenantConfJson = null;
+        String tenantConfLocation = CarbonUtils.getCarbonHome() + File.separator +
+                APIConstants.RESOURCE_FOLDER_LOCATION + File.separator +
+                APIConstants.API_TENANT_CONF;
+        File tenantConfFile = new File(tenantConfLocation);
+        byte[] data;
+        if (tenantConfFile.exists()) { // Load conf from resources directory in pack if it exists
+            FileInputStream fileInputStream = new FileInputStream(tenantConfFile);
```
Comment (dushaniw) [inline migration-client/wso2-api-migration-client/src/main/java/org/wso2/carbon/apimgt/migration/client/ScopeRoleMappingPopulationClient.java:402]: shall we close the inputstream or use try-with-resources

File: migration-client/wso2-api-migration-client/src/main/java/org/wso2/carbon/apimgt/migration/client/ScopeRoleMappingPopulationClient.java [hunk 2]
```diff
@@ -363,4 +383,40 @@ private String makePermissionsStringByEscapingSlash(String startPermission, Stri
...
+            data = IOUtils.toByteArray(fileInputStream);
+        } else { // Fallback to loading the conf that is stored at jar level if file does not exist in pack
+            InputStream inputStream =
```
Comment (dushaniw) [inline migration-client/wso2-api-migration-client/src/main/java/org/wso2/carbon/apimgt/migration/client/ScopeRoleMappingPopulationClient.java:405]: same here

File: migration-client/wso2-api-migration-client/src/main/java/org/wso2/carbon/apimgt/migration/client/ScopeRoleMappingPopulationClient.java (hunk 1 above)
Comment (msm1992) [reply migration-client/wso2-api-migration-client/src/main/java/org/wso2/carbon/apimgt/migration/client/ScopeRoleMappingPopulationClient.java]: fixed in https://github.com/wso2-extensions/apim-migration-resources/pull/47/commits/fb852a686e05b4603a7c1845fbb002595ac41962

File: migration-client/wso2-api-migration-client/src/main/java/org/wso2/carbon/apimgt/migration/client/ScopeRoleMappingPopulationClient.java (hunk 2 above)
Comment (msm1992) [reply migration-client/wso2-api-migration-client/src/main/java/org/wso2/carbon/apimgt/migration/client/ScopeRoleMappingPopulationClient.java]: fixed in https://github.com/wso2-extensions/apim-migration-resources/pull/47/commits/fb852a686e05b4603a7c1845fbb002595ac41962

**Pull request merged**